components/suggest/src/store.rs

```diff
@@ -486,10 +490,13 @@ where
         };
 
         let attachment_data = self.settings_client.get_attachment(&attachment.location)?;
-        match serde_json::from_slice::<SuggestAttachment<T>>(&attachment_data) {
-            Ok(attachment) => self.ingest_record(writer, record, |dao, record_id| {
-                ingestion_handler(dao, record_id, attachment.suggestions())
-            }),
+        match serde_json::from_slice::<OneOrMany<T>>(&attachment_data) {
+            Ok(attachments) => self.ingest_record(
```

Comment:
Nit: calling those `attachments` might confuse the readers, `attachment_payload` could be better IMHO.